src/Tasks/Unzip.cs

```diff
@@ -131,6 +131,14 @@ private void Extract(ZipArchive sourceArchive, DirectoryInfo destinationDirector
             {
                 FileInfo destinationPath = new FileInfo(Path.Combine(destinationDirectory.FullName, zipArchiveEntry.FullName));
 
+                // Zip archives can have directory entries listed explicitly; we should skip these.
+                // The parent directories of any file entries is created just before file creation.
+                char lastChar = destinationPath.FullName[destinationPath.FullName.Length - 1];
+                if (lastChar == Path.DirectorySeparatorChar || lastChar == Path.AltDirectorySeparatorChar)
```

Comment:
Is there a better way of checking whether the entry is a directory? In linux for example, you could have a file ending in backslash.

Reply:
This is what the .NET Framework does:

https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs#L109

```C#
if (Path.GetFileName(fileDestinationPath).Length == 0)
{
    // If it is a directory:

    if (source.Length != 0)
        throw new IOException(SR.IO_DirectoryNameWithData);

        Directory.CreateDirectory(fileDestinationPath);
}
else
{
    // If it is a file:
    // Create containing directory:
    Directory.CreateDirectory(Path.GetDirectoryName(fileDestinationPath));
    source.ExtractToFile(fileDestinationPath, overwrite: overwrite);
}
```

Reply:
I've replaced the check for 'is directory' with this one `Path.GetFileName(destinationPath.FullName).Length == 0`.

The directory is also now explicitly created as this allows archives containing empty directories to also be extracted fully.